tests/compiler/std/array.ts

```diff
@@ -1060,3 +1060,11 @@ __release(changetype<usize>(arr));
   let testArray: i32[][] = [[], []];
   assert(testArray.flat().length == 0);
 }
+
+// export extended arrays
+
+export class ArrayU32 extends Array<u32> {}
+export class ArrayU8 extends Array<u8> {}
+export class ArrayStr extends Array<string> {}
+// TODO:
+// export class ArrayArrayI32 extends Array<Array<i32>> {}
```

Comment:
What happens in this case?

Reply:
https://github.com/AssemblyScript/assemblyscript/pull/1353#issuecomment-649062944

But it's expected due to `Array` hasn't operator `<` or `>`

Reply:
I see, hmm. Perhaps if we had a better standard than `@operator`, like using unique symbols, one could check with `isDefined`. Looks like this isn't easily solvable right now and requires a follow-up PR anyway, though.

Reply:
I guess it could be solved something like this:
```ts
function comparator<T>(a : T, b: T): i32 {
   if (hasOperator<T>("<") && hasOperator<T>(">")) {
      return i32(a > b) - i32(a < b);
   } else {
      threw new TypeError("type hasn't comparison operators");
   }
}
```

Reply:
> Looks like this isn't easily solvable right now and requires a follow-up PR anyway, though.

Yes, definitely it's for separate PR

Reply:
Merged this so we have a fix for the reported use case already. Let's think about improving the other cases in a follow-up :)